planetiler-custommap/src/main/java/com/onthegomap/planetiler/custommap/configschema/TagCriteria.java

```diff
@@ -0,0 +1,60 @@
+package com.onthegomap.planetiler.custommap.configschema;
+
+import static com.onthegomap.planetiler.expression.Expression.matchAnyTyped;
+import static com.onthegomap.planetiler.expression.Expression.matchField;
+
+import com.onthegomap.planetiler.custommap.TagValueProducer;
+import com.onthegomap.planetiler.expression.Expression;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+
+public class TagCriteria extends HashMap<String, Object> {
```

Comment:
I try to avoid extending java collections, could we do this instead through a record that wraps a hashmap? Or provide the matcher function as a static method on a utility class?